### PR TITLE
Redesign company list page and add requested features to company detail

### PR DESCRIPTION
**Pull request by LudvigHz**
Redo the cards on the company list page and add features and changes as requested by Bedkom.

- [x] Add pagination to company events
- [x] Redisign company list cards
- [ ] Add filtering to company list
- [ ] Letter-navigation

### CODE REVIEW

**Comment by LudvigHz** [comment]
First draft:
Screenshot from 2019-03-18 20-41-35

Detail page:
Screenshot from 2019-03-18 20-42-29
Screenshot from 2019-03-18 20-42-42

**Comment by marilangaas** [comment]
Comments from Bedkom:

**Comment by marilangaas** [comment]
Comments from Bedkom: 
- Remove company name in text: All the companies have their names in their logos, so it is unnecessary to have the names in writing as well. 
- Add some colours to the icons.
- Description of the meaning of the icons at the top of the page. 

Wishes (not urgent!!):
- Filtering on companies that have active job announcements, events this semester etc. 
- Letter-navigation: if the student is looking for a company that starts with K, he/she can choose the letter from the alphabet and the site will navigate to the location of the "K"-companies.

**Comment by iselinee** [comment]
Updated with descriptions: 
Skjermbilde 2019-04-01 kl  22 06 41

**Comment by iselinee** [comment]
For mobile use:
The info-text got a bit long, so we shortened it to one paragraph, with a button to extend if wanted.

Skjermbilde 2019-04-01 kl  22 10 38

Skjermbilde 2019-04-01 kl  22 10 54

**Comment by iselinee** [comment]
Added a bit more color, and centered the viewMore button

Skjermbilde 2019-04-01 kl  23 47 03



Skjermbilde 2019-04-01 kl  23 47 17

**Comment by iselinee** [comment]
Skjermbilde 2019-09-04 kl  16 29 52

**Comment by kotlarz** [comment]
Waiting for approval from Bedkom.

**Comment by iselinee** [comment]
Bedkom has approved.

**Comment by JonasBak** [comment]
image

The stats are pushed down a bit by the picture because of a "height: 100%"

File: app/routes/company/components/CompaniesPage.js [hunk 1]
```diff
@@ -17,36 +21,47 @@ type Props = {
...
-            </span>
-          </div>
-        </Link>
-        {company.website && !company.website.includes('example.com') && (
-          <a href={company.website} rel="noopener noreferrer" target="_blank">
-            <div className={styles.website}>{websiteString}</div>
-          </a>
-        )}
+          </Flex>
+          {company.website && (
```
Comment (kotlarz) [inline app/routes/company/components/CompaniesPage.js]: This is nice! We should however have a URL validator on the website field for the Company model in the backend, even though we do a bit of validation on the frontend when editing companies. Nothing you can do about it here though.

File: app/routes/company/components/CompaniesPage.js [hunk 2]
```diff
@@ -64,19 +79,89 @@ const CompanyList = ({ companies = [] }: CompanyListProps) => (
...
+              er, les mer om hva de jobber med og se hvor de holder til. Bla deg
+              gjennom en oversikt over tidligere eller kommende arrangementer og
+              se hvem som har jobbannonser ute for øyeblikket. Hvis du vil lese
+              mer om bedriften så kan du navigere deg til nettsiden deres via
+              linken.
+            </p>
+
+            <p className={styles.infoText}>
+              Savner du en bedrift? Savner du noe informasjon om en bedrift? Ta
+              kontakt med Bedkom, vi tar gjerne imot innspill!
```
Comment (kotlarz) [inline app/routes/company/components/CompaniesPage.js]: Maybe add a link (or perhaps even better, mailto) to Bedkom?

File: app/routes/company/components/CompanyDetail.js [hunk 3]
```diff
@@ -80,6 +92,74 @@ const CompanyDetail = (props: Props) => {
...
+                  {viewOld ? 'Vis kun kommende' : 'Vis tidligere arrangementer'}
+                </button>
+              </th>
+            </tr>
+            {viewOld && eventTable(oldEvents)}
+          </tbody>
+        </table>
+        {viewOld && showFetchMoreEvents && (
+          <div
+            style={{
```
Comment (kotlarz) [inline app/routes/company/components/CompanyDetail.js]: Put this into the CSS file 😄

**Comment by kotlarz** [comment]
Also, have to fire whoever made the tables overflow everything on mobile.

image

**Pull request merged**